File: typescript/src/services/shares-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { CreateShareDto, PaginationOptions, ShareIdDto, ShareInfoDto, SharePageIdDto, UpdateShareDto } from '../models/index.js';


export class SharesApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async getShares(request: PaginationOptions): Promise<ApiResponse<unknown>> {
    return this.http.post(`shares`, request);
  }

  async getSharedPageInfo(request: ShareInfoDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`shares/page-info`, request);
  }

  async getShare(request: ShareIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`shares/info`, request);
  }

  async getShareForPage(request: SharePageIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`shares/for-page`, request);
  }

  async createShare(request: CreateShareDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`shares/create`, request);
  }

  async updateShare(request: UpdateShareDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`shares/update`, request);
  }

  async deleteShare(request: ShareIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`shares/delete`, request);
  }

  async getSharePageTree(request: ShareIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`shares/tree`, request);
  }

}

File: c#/src/Docmost.Sdk/Services/SharesApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class SharesApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> GetSharesAsync(PaginationOptions request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("shares", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetSharedPageInfoAsync(ShareInfoDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("shares/page-info", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetShareAsync(ShareIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("shares/info", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetShareForPageAsync(SharePageIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("shares/for-page", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> CreateShareAsync(CreateShareDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("shares/create", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> UpdateShareAsync(UpdateShareDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("shares/update", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> DeleteShareAsync(ShareIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("shares/delete", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetSharePageTreeAsync(ShareIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("shares/tree", request, cancellationToken).ConfigureAwait(false);
    }

}
